### DISCUSSION TITLE
Volume Mounting problems

### DISCUSSION THREAD

**Discussion by SteveChurch** [digitalocean/supabase-on-do#25] [Q&A] [answered]
I haven't touched my deployment in a while whilst my project has been in development, but I have just started deploying again and for some weird reason volume mounting doesn't seem to be happening, IE the persistent volume is always empty. 

I don't see any `mount` usages within the codebase, but feel like this previously worked fine? 

@timothymamo am I going mad? :-D

**Comment by SteveChurch** [comment]
I've thrown in a cheeky `sudo mount -o defaults,nofail,discard,noatime /dev/disk/by-id/blaa /mnt/blaa`

**Comment by timothymamo** [comment]
Hi @SteveChurch,

What's your deployment process looking like? You won't be able to mount the volume to another droplet whilst the volume is attached to another volume. The expected behaviour would be for the volume to be mounted to the droplet automagically. Are you getting any Warning/Error messages?

Obviously the `mount` command is a good workaround and if that works I'd say keep on creating :).

Tim

**Reply by SteveChurch** [reply]
@timothymamo for the most part it's similar to this base project. I'll spend some time soon to dig deeper, I'm honestly not sure what I may have changed that would effect the mounting process as this happens IIRC long before the cloud init scripts. 

🤷‍♂️ - I shall leave my manual mount in place for now as a fail safe 😀

**Reply by SteveChurch** [reply]
@timothymamo I have been looking through this again tonight and the DO docs show that although it's attached, the mounting isn't automated. I don't see anywhere where you actually mount the persistent volume to `/mnt/supabase_volume`

<img width="671" alt="image" src="https://user-images.githubusercontent.com/458008/228671465-98ec34be-fce3-4122-8e10-cc75267372dd.png">

https://docs.digitalocean.com/products/volumes/how-to/mount/

I can only assume that without the mounting / adding this to `fstab` that your version of this is not persisting data? 

That said, even with a mount command such as ```sudo mount -o defaults,nofail,discard,noatime /dev/disk/by-id/scsi-0DO_Volume_supabase-volume /mnt/supabase_volume``` everything weirdly resets so I must have something else going on 🤦🏻

**Reply by SteveChurch** [reply]
A cheeky 10 second sleep after the `mount` has fixed all my problems. :-D

**Reply by timothymamo** [reply]
Hi @SteveChurch, the mounting occurs as we specify the `volume_id` in the [droplet resource](https://github.com/digitalocean/supabase-on-do/blob/main/terraform/droplet.tf#L37). But you are right this only occurs with the initial instance that is created. For any redeployed instances the mounting is not automatically done for you (even though the volume is attached to the droplet).

Where did you specify the `mount`, and `sleep`, command within the terraform code? I've tried adding it as part of the `cloud_config` data source, but it's not working for me. Seeing as you've got it working would you mind creating a PR for this?

Thanks again for testing out the repo and giving amazing feedback,

Tim

**Reply by SteveChurch** [reply]
@timothymamo I have also added it as part of the `cloud_config` but you first need to `mkdir -p /mnt/supabase_volume` here it my config. 

```
    part {
        content_type = "text/x-shellscript"
        filename     = "init.sh"
        content      = <<-EOF
            #!/bin/bash
            mkdir -p /mnt/supabase_volume
            sudo mount -o defaults,nofail,discard,noatime /dev/disk/by-id/scsi-0DO_Volume_supabase-volume /mnt/supabase_volume
            sleep 10
            mkdir -p /mnt/supabase_volume/supabase/data
            cd /root/supabase
            /usr/bin/docker compose -f /root/supabase/docker-compose.yml up -d
            EOF
    }
    ```

**Reply by timothymamo** [reply]
Ahhh you added it as part of the original `shellscript`. I had it as a separate script, but your placement makes a lot more sense 🤦 (unsure how cloud config deals with order).

I'll add it to the terraform code and update it. Again thanks for debugging the system 😃 .

**Reply by SteveChurch** [reply]
Yeah, I did toy with that idea but also had the same questions about ordering. 

No problem at all. The project I have used supabase for went live last night! :-D - You've helped me a lot with this repo.

**Reply by timothymamo** [reply]
Awesome to hear!! If you're happy to do so, share your project here and I'll see how we can give it some DO Love 😄.

**Reply by SteveChurch** [reply]
Sure! https://dislist.gg is the project. 

It's a mixture of Supabase self hosted, App framework with SLS functions and a static site. It's been a really fun project to work on. Happy to be part of a case study if you like ;-)

**Comment by SteveChurch** [comment, marked as answer]
https://github.com/digitalocean/supabase-on-do/discussions/25#discussioncomment-5477106
